File: solutions/cpp/series/1/series.cpp
#include "series.h"
#include <stdexcept>

namespace series
{
    std::vector<std::string> slice(const std::string &input, int k)
    {
        if (k > static_cast<int>(input.size()))
            throw std::domain_error("slice length too large");
        if (k <= 0)
            throw std::domain_error("slice length can't be zero or negative");

        std::vector<std::string> result;
        for (size_t i = 0; i <= input.size() - k; ++i)
            result.push_back(input.substr(i, k));
        return result;
    }
} // namespace series

File: solutions/java/series/1/src/main/java/Series.java
import java.util.ArrayList;
import java.util.List;

public class Series {
    private final String input;

    public Series(String input) {
        if (input == null || input.isEmpty()) {
            throw new IllegalArgumentException("series cannot be empty");
        }
        this.input = input;
    }

    public List<String> slices(int sliceLength) {
        if (sliceLength <= 0) {
            throw new IllegalArgumentException("slice length cannot be negative or zero");
        }
        if (sliceLength > input.length()) {
            throw new IllegalArgumentException("slice length cannot be greater than series length");
        }

        List<String> result = new ArrayList<>();
        for (int i = 0; i <= input.length() - sliceLength; i++) {
            result.add(input.substring(i, i + sliceLength));
        }
        return result;
    }
}